platform/viewer/src/routes/Mode/Mode.tsx

```diff
@@ -131,6 +131,34 @@ export default function ModeRoute({
     hotkeys: hotkeyObj,
     hangingProtocol,
   } = mode;
+
+  const runTimeHangingProtocolId = searchParams.get('hangingprotocolid');
+  const token = searchParams.get('token');
+
+  if (token) {
```

Comment:
Would not this be better done in a `useEffect`?

Reply:
since there is no state update internally I don't think so